### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by danieljak**
Map dependency with FOUND variable. Some boolean expression support.

### CODE REVIEW

File: tools/cmake2meson.py [hunk 1]
```diff
@@ -143,6 +143,56 @@ def __init__(self, cmake_root):
...
+        dependency = arg[0:arg.find('_FOUND')].lower()
+        if ('_FOUND' in arg) and any(dependency in d for d in self.dependencies):
+            for d in self.dependencies:
+                if d in dependency or dependency in d:
+                    return d
+            return dependency
+        if dependency in self.dependencyMap:
+            return self.dependencyMap[dependency]
+        return False
+
```
Comment (ignatenkobrain) [inline tools/cmake2meson.py]: __[flake8]__
* [W293] blank line contains whitespace

File: tools/cmake2meson.py [hunk 2]
```diff
@@ -143,6 +143,56 @@ def __init__(self, cmake_root):
...
+            booleanExpr = True
+        elif value == "'STRGREATER'":
+            value = " > "
+            booleanExpr = True
+        elif value == "'MSVC'":
+            value = "meson.get_compiler('cpp').get_id() == 'msvc'"
+        # Not implemented: MATCHES
+        params[0] = value
+        params[1] = booleanExpr
+        return value;
```
Comment (ignatenkobrain) [inline tools/cmake2meson.py]: __[flake8]__
* [E703] statement ends with a semicolon

File: tools/cmake2meson.py [hunk 3]
```diff
@@ -163,9 +222,21 @@ def convert_args(self, args, as_array=True):
                 print(i)
                 raise RuntimeError('Unknown arg type.')
         if len(res) > 1:
-            return start + ', '.join(res) + end
+            subres = []
+            booleanExpr = False
+            for i, val in enumerate(res):
+                params = [res[i], booleanExpr]
+                res[i] = self.get_translated_value(params);
```
Comment (ignatenkobrain) [inline tools/cmake2meson.py]: __[flake8]__
* [E703] statement ends with a semicolon

File: tools/cmake2meson.py [hunk 4]
```diff
@@ -163,9 +222,21 @@ def convert_args(self, args, as_array=True):
...
+                    booleanExpr = True
+            if booleanExpr:
+                subres.append(' '.join(res))
+            else:
+                subres.append(', '.join(res))
+            return start + ''.join(subres) + end
         if len(res) == 1:
-            return res[0]
+            params = [res[0], False]
+            return self.get_translated_value(params);
```
Comment (ignatenkobrain) [inline tools/cmake2meson.py]: __[flake8]__
* [E703] statement ends with a semicolon

File: tools/cmake2meson.py [hunk 5]
```diff
@@ -179,13 +250,15 @@ def write_entry(self, outfile, t):
             line = "subdir('" + t.args[0].value + "')"
         elif t.name == 'pkg_search_module' or t.name == 'pkg_search_modules':
             varname = t.args[0].value.lower()
+            #self.dependencies.append(i.value.lower() for i in t.args[1:])
```
Comment (ignatenkobrain) [inline tools/cmake2meson.py]: __[flake8]__
* [E265] block comment should start with '# '

**Comment by jpakkane** [comment]
What is this MR supposed to achieve? The description does not seem to match the code at all.

**Comment by danieljak** [comment]
I'm new as a github contributer, so apologies if I'm messing things up..

I was converting a cmake project to meson and added some more stuff to automatically put it correctly in the meson.build file. I ended up doing it manually anyway, but since I did some of it I thought I'd share those changes;
I mapped dependencies with FOUND variable. Ie: 
```
find_package(OpenSSL REQUIRED)
if(OPENSSL_FOUND)
```
..would give:
```
openssl_dep = dependency('openssl')
if openssl_dep.found()
```

And
```
find_package(APR)
find_package(Apache2)
if(APRUTIL_FOUND AND APACHE_FOUND)
```
..would give:
```
apr_dep = dependency('apr')
apache2_dep = dependency('apache2')
if apr_dep.found() and apache2_dep.found()
```

The other thing was some boolean expression support. Now this code got uglier and there are surely better ways to solve this, but it works.
So the same if as above:
```
if(APRUTIL_FOUND AND APACHE_FOUND)
```
..correctly adds the 'and' without quotes and comma.

The latter commits were corrections on the feedback from ignatenkobrain.

**Comment by jpakkane** [comment]
Updating the CMake importer is ok but it should not change stuff in other parts, such as this in ninjabackend.py

    commands += self.build.get_global_post_link_args(linker)

Any new features must be first reviewed standalone.

In addition things like this:

    find_package(OpenSSL REQUIRED)
    if(OPENSSL_FOUND)

don't make sense because the if branch will always evaluate to true.

We have not put that much effort into the CMake converter because experience has shown that CMake build definitions are almost always messy and depend on weird stuff. A fair bit of manual work is always required so the point of diminishing returns comes up quite quickly.

**Comment by danieljak** [comment]
Right, I'm not sure why the other changes were included in this request.

That was a different commit, not meant to be part of the pull request.
It adds the ability to add linker args at the end, which was necessary to make a shared_library to compile.

**Comment by jpakkane** [comment]
Closing this to reduce clutter on the MR page. Feel free to open a new MR with a clean version. Thanks.

**Pull request closed without merging**